lib/relations.js

```diff
@@ -358,6 +361,9 @@ class Relations {
                             return existingTargets.detach(targetToDetach, _.pick(opts, ['transacting']));
                         });
                     }).then(() => {
+                        if (model.hooks && model.hooks.belongsToMany && model.hooks.belongsToMany.after) {
```

Comment:
I don't think we should change this... but I really dislike the naming inconsistency here. Why one hook is `after` and another is super explicit `beforeRelationCreation` 😿  Maybe... since we are going to use this moving on let's add an alias to one of these, I'd vote for `beforeRelationCreation = before`, and use that? Wdyt?

Reply:
Yeah totally agree!